### PR TITLE
Aha/checks

### PR DESCRIPTION
**Pull request by andyHa**

### CODE REVIEW

File: src/main/java/sirius/biz/tenants/UserAccountData.java [hunk 1]
```diff
@@ -110,28 +109,35 @@ public UserAccountData(BaseEntity<?> userObject) {
...
-                            .handle();
-        }
 
-        userObject.assertUnique(UserAccount.USER_ACCOUNT_DATA.inner(LOGIN).inner(LoginData.USERNAME),
-                                getLogin().getUsername());
+        // Ensure that the username is is filled and unique...
+        userObject.verifyIfChangedFailIfEmpty(UserAccount.USER_ACCOUNT_DATA.inner(LOGIN).inner(LoginData.USERNAME),
+                                              () -> {
+                                                  // Make it lowercase...
+                                                  getLogin().setUsername(getLogin().getUsername().toLowerCase());
```
Comment (idlira) [inline src/main/java/sirius/biz/tenants/UserAccountData.java]: Requires a check if already saved data is lowercase (and a batch-convert) when not?

File: src/main/java/sirius/biz/tenants/UserAccountData.java [hunk 2]
```diff
@@ -110,28 +109,35 @@ public UserAccountData(BaseEntity<?> userObject) {
...
-        } else {
-            throw Exceptions.createHandled()
-                            .withNLSKey("Property.fieldNotNullable")
-                            .set("field", NLS.get("LoginData.username"))
-                            .handle();
-        }
 
-        userObject.assertUnique(UserAccount.USER_ACCOUNT_DATA.inner(LOGIN).inner(LoginData.USERNAME),
-                                getLogin().getUsername());
+        // Ensure that the username is is filled and unique...
```
Comment (idlira) [inline src/main/java/sirius/biz/tenants/UserAccountData.java]:
```suggestion
        // Ensure that the username is filled and unique...
```

File: src/main/java/sirius/biz/model/ContactData.java [hunk 3]
```diff
@@ -24,21 +22,24 @@
...
     /**
-     * Validates a phone numner.
+     * Validates a phone number.
      */
-    public static final Pattern VALID_PHONE_NUMBER =
-            Pattern.compile("\\+?\\d+( \\d+)*( */( *\\d+)+)?( *\\-( *\\d+)+)?");
-
-    @Transient
-    private boolean validatePhoneNumbers;
+    public static final Pattern VALID_PHONE_NUMBER = Pattern.compile("\\+?\\d+( \\d+)*( */( *\\d+)+)?( *-( *\\d+)+)?");
```
Comment (idlira) [inline src/main/java/sirius/biz/model/ContactData.java]: Is this format valid internationally?

File: src/main/java/sirius/biz/tenants/UserAccountData.java (hunk 1 above)
Comment (andyHa) [reply src/main/java/sirius/biz/tenants/UserAccountData.java]: this code isn't new, it was just moved to another place...

File: src/main/java/sirius/biz/model/ContactData.java (hunk 3 above)
Comment (andyHa) [reply src/main/java/sirius/biz/model/ContactData.java]: This regex isn't new / has not changed semantically. The \\- was a redundant escape which has been removed (as suggested by IntelliJ) - hopefully the regex was/is valid :)

**Pull request merged**